sqlglot/optimizer/simplify.py

```diff
@@ -854,15 +846,27 @@ def cast_value(value: t.Any, to: exp.DataType) -> t.Optional[t.Union[datetime.da
     return None
 
 
-def extract_date(cast: exp.Cast) -> t.Optional[t.Union[datetime.date, datetime.date]]:
+def extract_date(cast: exp.Expression) -> t.Optional[t.Union[datetime.date, datetime.date]]:
+    to: exp.DataType
+    if isinstance(cast, exp.Cast):
+        to = cast.to
+    elif isinstance(cast, exp.TsOrDsToDate):
+        to = exp.DataType.build(exp.DataType.Type.DATE)
```

Comment:
Can also do `build("date)"`

Reply:
Why? that seems slower, given it requires parsing. And in this case, we know what type we want.

Reply:
Fair enough, I just feel part of that API was to reduce the visual noise of writing the fqn of enums, but don't feel strongly about it